data/tutorials/language/3ds_06_options.md

```diff
@@ -0,0 +1,167 @@
+---
+id: options
+title: Options
+description: >
+  Add nothing-as-value to anything to avoid confusion between something and “no such thing“.
+category: "Introduction"
+---
+
+# Options
+
+## Introduction
+
+An option is a value that wraps another value or nothing, if there isn't anything to wrap. The predefined type `option` is the 
+<!-- $MDX non-deterministic=command -->
+```ocaml
+# #show option;;
+type 'a option = None | Some of 'a
+```
+
+Here is 42, stored inside an `option` using the data carrying constructor
+`Some`:
+
+```ocaml
+# Some 42;;
+- : int option = Some 42
+```
+
+The `None` constructor means no data is available.
+
+In other words, a value of type `t option` for some type `t` represents
+* either a value `v` of type `t`, wrapped as `Some v`
+* no such value, then `o` has the value `None`
+
+The option type is very useful when lack of data is better handled as a special value (_i.e.,_ `None`) rather than an exception. It is the type-safe version of returning error values such as in C, for instance. Since no data has any special meaning, confusion between regular values and absence of value is impossible. In computer science, this type is called the [option
```

Comment:
```suggestion
The option type is very useful when lack of data is better handled as a special value (_i.e.,_ `None`) rather than an exception. It is the type-safe version of returning error values such as in C, for instance. Since no data has any special meaning, confusion between regular values and the absence of value is impossible. In computer science, this type is called the [option
```